ucsmsdk/utils/comparesyncmo.py

```diff
@@ -0,0 +1,489 @@
+
+        # comparing xtra property
+        from_xtra_props = from_mo._ManagedObject__xtra_props
+        to_xtra_props = to_mo._ManagedObject__xtra_props
+        if from_xtra_props:
+            for prop in from_xtra_props:
+                if prop in to_xtra_props and from_xtra_props[prop].value != \
+                        to_xtra_props[prop].value:
+                    # diff.append(prop)
+                    UcsWarning("Ignoring xtra property '%s' of '%s'" % (
+                        prop, from_mo.dn))
+
+    if len(diff) > 0:
+        return _CompareStatus.PROPS_DIFFERENT
+
+    return _CompareStatus.EQUAL
+
+
+def _update_mo_dn_along_with_naming_properties(mo, ref_dn):
+    """
+    Internal method to modify the naming properties of mo using dn.
+    """
+
+    # modify diffmo dn with refmo dn
+    mo.__dict__['dn'] = ref_dn
+
+    # modify diffmo rn with refmo rn
+    ref_rn = re.sub(r'^.*/', '', ref_dn)
+    mo.__dict__['rn'] = ref_rn
+
+    # modify diff mo naming properties using ref_rn
+    naming_prop_dict = ucscoreutils.get_naming_props(rn_str=ref_rn,
+                                                     rn_pattern=mo.mo_meta.rn)
+    for prop in naming_prop_dict:
+        mo.__dict__[prop] = naming_prop_dict[prop]
+
+
+def _translate_managed_object(mo, xlate_org, xlate_map):
+    """
+    Internal method used to translate a managed object which helps in comparing
+    two mo with different dn.
+    """
+
+    ref_dn = None
+    if xlate_org is not None:
+        match_obj = re.match(
+            r'^(org-[\-\.:_a-zA-Z0-9]{1,16}/)*org-[\-\.:_a-zA-Z0-9]{1,16}',
+            mo.dn)
+        if match_obj:
+            if mo.get_class_id() == "OrgOrg":
+                ref_dn = re.sub("%s" % (match_obj.group(0)),
+                                "%s" % xlate_org,
+                                mo.dn)
+            else:
+                ref_dn = re.sub("^%s/" % (match_obj.group(0)),
+                                "%s/" % xlate_org,
+                                mo.dn)
+            _update_mo_dn_along_with_naming_properties(mo, ref_dn)
+
+    if xlate_map is not None:
+        diff_dn = mo.dn
+        if diff_dn in xlate_map:
+            ref_dn = xlate_map[diff_dn]
+        else:
+            diff_dn = re.sub(r'[/]*[^/]+$', '', diff_dn)
+            while diff_dn is not None or diff_dn == "":
+                if diff_dn not in xlate_map:
+                    diff_dn = re.sub(r'[/]*[^/]+$', '', diff_dn)
+                    continue
+
+                ref_dn = re.sub("^%s/" % diff_dn,
+                                "%s/" % xlate_map[diff_dn],
+                                mo.dn)
+                break
+        _update_mo_dn_along_with_naming_properties(mo, ref_dn)
+
+    return mo
+
+
+def write_mo_diff(diff_obj):
+    """
+    Writes the difference managedObject(output of CompareManagedObject)
+    on the terminal.
+    """
+
+    tab_size = 8
+    if isinstance(diff_obj, list) and len(diff_obj) > 0:
```

Comment:
@ragupta-git for code clarity it is better to try and reduce levels of indentation for the regular code path when applicable. this could be re-written as 
```
if not (condition):
    return
rest code of the code can start at the upper level again.
```